tl/features/context_match.py

```diff
@@ -0,0 +1,332 @@
+        new_sent = self.tokenize(new_sent)
+        return new_sent
+
+    def tokenize(self, sent):
+        sent = sent.split(" ")
+        return sent
+
+    def remove_punctuation(self, sent):
+        res = re.sub(r'[^\w\s]', '', sent)
+        return res
+
+    def quantity_matching(self, context_target, context_try):
+        if context_target == 0.0 and context_try == 0.0:
+            return 1
+        max_val = max(abs(context_target), abs(context_try))
+        abs_diff = abs(context_target - context_try)
+        final_val = 1 - (abs_diff/max_val)
+        return final_val
+
+    def remove_comma(self, string):
+        string = string.replace(',', '')
+        return string
+
+    def matching_with_quantity(self, context_to_do, all_props, check = "q"):
+        res = [idx for idx in all_props if idx.lower().startswith(check.lower())]
+        prop_val = "NaN"
+        sim = 0.0
+        res2 = "NaN"
+        for x in res:
+            x = x.split(":")
+            n = x[0]
+            check_with_t = n[1:]
+            check_with = float(check_with_t.replace('"', ''))
+            check_for = float(context_to_do.replace('"', ''))
+            value = self.quantity_matching(check_with, check_for)
+            if value >= self.threshold_2 and value >= sim:
+                prop_val = x[1] 
+                sim = value
+                res2 = check_with
+        #print(prop_val, sim, value)
+        sim = round(sim, 4)
+        return prop_val, sim
+
+    def matching_with_date(self, context_to_do, all_props, check = "d"):
+        res = [idx for idx in all_props if idx.lower().startswith(check.lower())]
+        prop_val = "NaN"
+        sim = 0.0
+        for x in res:
+            x = x.split(":")
+            n = x[0]
+            check_with = self.remove_punctuation(n[1:])
+            if self.remove_punctuation(context_to_do) == check_with:
+                prop_val = x[1] 
+                sim = 1.0
+        return prop_val, sim
+
+    def same_elements(self, lst):
+        ele = lst[0]
+        chk = True
+        for item in lst:
+            if ele != item:
+                chk = False
+                break;         
+        return chk
+
+    def matching_with_item(self, context_to_do, all_props, check = "i"): 
+        p_context_to_do = self.preprocess(context_to_do)
+        res = [idx for idx in all_props if idx.lower().startswith(check.lower())]
+        prop_val = "NaN"
+        max = 0.0
+        for x in res:
+            x = x.split(":")
+            n = x[0]
+            check_with = self.remove_punctuation(n[1:])
+            p_check_with = self.preprocess(check_with)
+            sim = self.symmetric_monge_elkan(p_check_with, p_context_to_do)
+            if sim >= self.threshold_1 and sim >= max:
+                if len(x) > 1:#Resolves error if the context does not have a property
+                    prop_val = x[1]
+                    max = sim
+        return prop_val, max
+
+    def Convert(string):

```

Comment:
method names should be all lower cases. Also this method is not required, please remove